packages/Takeover/src/components/EscListener.js

```diff
@@ -0,0 +1,33 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+const propTypes = {
+  /** Callback triggered on ESC key press */
+  on: PropTypes.func.isRequired,
+};
+
+const EscListener = ({ on, shouldStopEscapePropagation }) => {
```

Comment:
To me, this seems more a hook than a component, I think @alexzherdev commented about this in a previous PR.